src/id-api.js

```diff
@@ -0,0 +1,136 @@
+"use strict";
+const Signature = require('./signature');
+const https = require('https');
+
+class IDApi {
+
+  constructor(partner_id, api_key, sid_server) {
+    this.partner_id = partner_id;
+    this.sid_server = sid_server;
+    this.api_key = api_key;
+
+    if (['0', '1'].indexOf(sid_server.toString()) > -1) {
+      var sid_server_mapping = {
+        '0': '3eydmgh10d.execute-api.us-west-2.amazonaws.com/test',
+        '1': 'la7am6gdm8.execute-api.us-west-2.amazonaws.com/prod'
+      };
+      this.url = sid_server_mapping[sid_server.toString()];
+    } else {
+      this.url = sid_server;
+    }
+  }
+
+  submit_job(partner_params, id_info) {
+    var _private = {
+      data: {
+        timestamp: Date.now(),
+        url: this.url,
+        partner_id: this.partner_id,
+        api_key: this.api_key,
+        sid_server: this.sid_server
+      },
+      validateInputs: function() {
+        _private.partnerParams(partner_params);
+        _private.idInfo(id_info);
+      },
+      partnerParams: function(partnerParams) {
+        if (!partnerParams) {
+          throw new Error('Please ensure that you send through partner params');
+        }
+
+        if (typeof partnerParams !== 'object') {
+          throw new Error('Partner params needs to be an object');
+        }
+
+        ['user_id', 'job_id', 'job_type'].forEach((key) => {
+          if (!partnerParams[key] || (typeof partnerParams === 'string' && partnerParams[key].length === 0)) {
+            throw new Error(`Please make sure that ${key} is included in the partner params`);
+          }
+        });
+
+        partnerParams['job_type'] = parseInt(partnerParams['job_type'], 10);
+        if (parseInt(partnerParams['job_type'], 10) !== 5) {
+          throw new Error('Please ensure that you are setting your job_type to 5 to query ID Api');
+        }
+
+        _private.data.partner_params = partnerParams;
+      },
+      idInfo: function(idInfo) {
```

Comment:
I think we can safely add a validation that id_number is present and has length also. what do you think about that?